Assets/Scripts/SceneLoader.cs

```diff
@@ -0,0 +1,70 @@
+using Sirenix.OdinInspector;
+using System;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Celeritas
+{
+	public class SceneLoader : MonoBehaviour
+	{
+		[Flags]
+		public enum Scene
+		{
+			Persistent = 1 << 1,
+			Mainmenu = 1 << 2,
+			Main = 1 << 3,
+			RunStart = 1 << 4,
+		}
+
+		[SerializeField]
+		private Scene scene;
+
+		[SerializeField]
+		private LoadSceneMode mode;
+
+		[SerializeField, InfoBox("Load on awake only works in builds, does not occur in editor. Destroys component")]
+		private bool loadOnAwake;
+
+		private void Awake()
+		{
+			if (loadOnAwake)
+			{
+#if !UNITY_EDITOR
+			LoadScene();
+#endif
+				Destroy(this);
+			}
+		}
+
+		public void LoadScene()
```

Comment:
Once again, not necessary, but would be nice to have a quick comment describing what this public method does. Especially for people wanting to quickly understand how it all works.